Relay/Connection/Paginator.php

```diff
@@ -61,13 +61,19 @@ public function forward($args)
         $limit = $args['first'];
         $offset = ConnectionBuilder::getOffsetWithDefault($args['after'], 0);
 
-        // The extra fetched element is here to determine if there is a next page.
-        $entities = call_user_func($this->fetcher, $offset, $limit + 1);
+        // If we don't have a cursor or if it's not valid, then we must not use the slice method
+        if (!is_numeric(ConnectionBuilder::cursorToOffset($args['after'])) or !$args['after']) {
+            $entities = call_user_func($this->fetcher, $offset, $limit + 1);
 
-        return ConnectionBuilder::connectionFromArraySlice($entities, $args, [
-            'sliceStart' => $offset,
-            'arrayLength' => $offset + count($entities),
-        ]);
+            return ConnectionBuilder::connectionFromArray($entities, $args);
+        } else {
+            $entities = call_user_func($this->fetcher, $offset, $limit + 2);
```

Comment:
I'll explain in the doc. 
Basically Relay requiere that the sample you provide contains the item before the slice you want, and the item after in order to compute the 'hasNextPage' meta. 
Exemple:
The full data set is: a b c d e
I want 2 items after b, meaning a limit of 2
The decoded cursor will be 1
If we use $limit + 1, the slice returned by my data provider will be 'b, c, d' and Relay will return 'c, d' which is correct, but 'hasNextPage' will be false since we are not telling Relay that there's something after 'd'